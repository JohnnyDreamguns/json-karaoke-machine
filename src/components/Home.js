import React, { useEffect } from 'react';
import '../styles.css';
import { togglePlay } from '../includes/music-machine';
import { default as song } from '../songs/VanHalenJump';
import Header from './Header';
import SongChooser from './SongChooser';
import Drums from './Drums';
import Bass from './Bass';
import PolySynth from './PolySynth';
import Footer from './Footer';
import Lyrics from './Lyrics';

const Home = ({
  initHomePage,
  tempo,
  beatNumber,
  handleChangeTempo,
  isPlaying
}) => {
  useEffect(() => {
    initHomePage();
  }, []);

  return (
    <div className="App">
      <Header
        play={togglePlay(isPlaying)}
        isPlaying={isPlaying}
        tempo={tempo}
        handleChangeTempo={handleChangeTempo}
      />
      <SongChooser
        play={togglePlay(isPlaying)}
        isPlaying={isPlaying}
        tempo={tempo}
        handleChangeTempo={handleChangeTempo}
      />
      <Drums data={song.drums} beatNumber={beatNumber} />
      <Bass data={song.bass} beatNumber={beatNumber} />
      <PolySynth data={song.polySynth} beatNumber={beatNumber} />
      <Footer />
      {isPlaying && <Lyrics data={song.lyrics} beatNumber={beatNumber} />}
    </div>
  );
};

export default Home;
