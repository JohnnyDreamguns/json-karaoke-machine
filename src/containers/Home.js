import { connect } from 'react-redux';
import '../styles.css';
import { setTempo } from '../actions/index';
import Home from '../components/Home';
import { init } from '../includes/music-machine';
import { default as song } from '../songs/VanHalenJump';

const initHomePage = () => {
  init(song);
};

export const mapStateToProps = state => ({
  initHomePage: initHomePage,
  tempo: state.settings.tempo,
  beatNumber: state.settings.beatNumber,
  isPlaying: state.settings.isPlaying,
  currentLine: state.settings.currentLine
});

export const mapDispatchToProps = dispatch => ({
  handleChangeTempo: ({ target: { value } }) => {
    dispatch(setTempo(value));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Home);
