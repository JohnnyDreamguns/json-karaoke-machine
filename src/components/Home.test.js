import 'babel-polyfill';
import React from 'react';
import Home from './Home';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { mount } from 'enzyme';

const defaultProps = {
  togglePlay: jest.fn(),
  isPlaying: true,
  tempo: 120,
  handleChangeTempo: jest.fn(),
  beatNumber: 20
};

describe('Home', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Home {...defaultProps} />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls initHomePage on render', () => {
    const mockInitHomePage = jest.fn();
    mount(
      <Router>
        <Home {...defaultProps} initHomePage={mockInitHomePage} />
      </Router>
    );
    expect(mockInitHomePage.mock.calls.length).toBe(1);
  });
});
