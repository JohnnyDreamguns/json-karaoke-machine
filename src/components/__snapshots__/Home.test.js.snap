// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  className="App"
>
  <nav
    className="navbar navbar-expand-lg navbar-light fixed-top"
    id="mainNav"
  >
    <div
      className="container"
    >
      <nav>
        <a
          className="navbar-brand js-scroll-trigger"
          href="/"
          onClick={[Function]}
        >
          JSON Karaoke Machine
        </a>
        <span>
          |
        </span>
        <a
          className="navbar-brand js-scroll-trigger"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </nav>
      <div
        className="play-controls"
      >
        <div>
          <button
            className="btn-no-sound"
            id="btn-no-sound"
            onClick={[Function]}
          >
            No Sound?
          </button>
          <input
            id="isVisible"
            type="hidden"
            value="hidden"
          />
        </div>
        <button
          aria-checked="false"
          className="btn btn-outline btn-xl"
          data-playing="false"
          onClick={[Function]}
          role="switch"
        >
          Stop
        </button>
        <div
          className="tempo-control"
        >
          <label
            forhtml="tempo"
          >
            120
            bpm
          </label>
          <input
            id="tempo"
            max="250"
            min="70"
            onChange={[MockFunction]}
            step="1"
            type="range"
            value={120}
          />
        </div>
      </div>
    </div>
  </nav>
  <header
    className="masthead"
  >
    <div
      className="container h-100"
    >
      <div
        className="row h-100"
      >
        <div
          className="col-lg-7 my-auto"
        >
          <div
            className="header-content mx-auto"
          >
            <h1
              className="mb-5"
            >
              Press play to hear the sweet sounds of the Web Audio API
            </h1>
            <div
              className="header-controls"
            >
              <button
                aria-checked="false"
                className="btn btn-outline btn-xl"
                data-playing="false"
                onClick={[Function]}
                role="switch"
              >
                Stop
              </button>
              <div
                className="tempo-control"
              >
                <label
                  forhtml="tempo"
                >
                  120
                  bpm
                </label>
                <input
                  id="tempo"
                  max="250"
                  min="70"
                  onChange={[MockFunction]}
                  step="1"
                  type="range"
                  value={120}
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-lg-5 my-auto"
        >
          <h3>
            Now Playing: Jump by Van Halen
          </h3>
          <img
            alt=""
            className="img-fluid"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
    </div>
  </header>
  <section
    className="bg-primary text-center instrument"
  >
    <h2>
      Drums
    </h2>
    <div
      className="json-display"
    >
      <div
        className="json"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "[<br><span class=\\"curly-brace\\">{</span>  <span class=\\"key\\">\\"type\\":</span> <span class=\\"value\\">\\"snare\\"</span> <span class=\\"curly-brace\\">}</span><br>]",
            }
          }
        />
      </div>
      <div>
        <img
          alt=""
          className="img-fluid"
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </div>
    </div>
  </section>
  <section
    className="features instrument"
  >
    <h2>
      Bass
    </h2>
    <div
      className="json-display"
    >
      <div
        className="json"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "[<br><span class=\\"curly-brace\\">{</span>  <span class=\\"key\\">\\"note\\":</span> <span class=\\"value\\">261.6</span>, <span class=\\"key\\">\\"octave\\":</span> <span class=\\"value\\">-1</span>, <span class=\\"key\\">\\"duration\\":</span> <span class=\\"value\\">1</span> <span class=\\"curly-brace\\">}</span>,<br><span class=\\"curly-brace\\">{</span>  <span class=\\"key\\">\\"note\\":</span> <span class=\\"value\\">261.6</span>, <span class=\\"key\\">\\"octave\\":</span> <span class=\\"value\\">-2</span>, <span class=\\"key\\">\\"duration\\":</span> <span class=\\"value\\">1</span> <span class=\\"curly-brace\\">}</span><br>]",
            }
          }
        />
      </div>
      <div>
        <img
          alt=""
          className="img-fluid"
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </div>
    </div>
  </section>
  <section
    className="bg-primary text-center instrument"
  >
    <h2>
      PolySynth
    </h2>
    <div
      className="json-display"
    >
      <div
        className="json"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "[<br><br><br><br>]",
            }
          }
        />
      </div>
      <div>
        <img
          alt=""
          className="img-fluid"
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </div>
    </div>
  </section>
  <footer>
    <div
      className="container"
    >
      <p>
        Â© JSON Karaoke Machine 2019. All Rights Reserved.
      </p>
    </div>
  </footer>
  <div
    className="lyrics"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
    />
  </div>
</div>
`;
